frontend/catalyst/python_interface/inspection/specs.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains the implementation of the `specs` function for the Unified Compiler."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Literal
+
+from ..compiler import Compiler
+from .specs_collector import ResourcesResult, specs_collect
+from .xdsl_conversion import get_mlir_module
+
+if TYPE_CHECKING:
+    from catalyst.jit import QJIT
+
+
+class StopCompilation(Exception):
+    """Custom exception to stop compilation early when the desired specs level is reached."""
+
+
+def mlir_specs(
+    qnode: QJIT, level: int | tuple[int] | list[int] | Literal["all"], *args, **kwargs
+) -> ResourcesResult | dict[str, ResourcesResult]:
```

Comment:
I was confused initially because this signature implies that we're doing away with the wrapper-style implementation. However, this will be "private" and eventually accessed through `qml.specs`, so I see why you've implemented it this way.